llvm/test/CodeGen/NVPTX/bug21465.ll

```diff
@@ -12,7 +12,7 @@ define ptx_kernel void @_Z11TakesStruct1SPi(ptr byval(%struct.S) nocapture reado
 entry:
 ; CHECK-LABEL: @_Z11TakesStruct1SPi
 ; PTX-LABEL: .visible .entry _Z11TakesStruct1SPi(
-; CHECK: addrspacecast ptr %input to ptr addrspace(101)
+; CHECK: call ptr addrspace(101) @llvm.nvvm.internal.noop.addrspacecast.p101.p0(ptr %input)
```

Comment:
Naming nit: I think `noop.addrspacecast` does not quite match what we want to do here. It's not that we're actually casting the pointer. It's more that we're letting LLVM know the actual AS of the argument which happens to be represented by a generic pointer. 

So, technically it does look like a no-op cast (changes AS, does not actually do anything). I think "wrapped" that you used in the description would be a better name here. How about `llvm.nvvm.internal.addrspace.wrap` ?

Reply:
Sure, I don't have very strong feelings on the name either way. I've updated to use this name.